packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -124,37 +203,115 @@ export function Tab<T>(props: TabProps<T>) {
+    selectedTab,
+    // selectedKey is provided so that the TabLine styles are updated when the TabPicker's width updates from a selection change
+    selectedKey
+  } = props;
+
   let verticalSelectionIndicatorOffset = 12;
   let {direction} = useLocale();
   let {scale} = useProvider();
 
+  // TODO: Figure out the best way to get rid of the wiggle when it goes from non-collapsed to collapsed
+  // Only way I've found is to initialize the state with the calulated transform...
   let [style, setStyle] = useState({
     width: undefined,
     height: undefined,
     transform: undefined
   });
 
   useLayoutEffect(() => {
-    let styleObj = {transform: undefined, width: undefined, height: undefined};
-    styleObj.transform = orientation === 'vertical'
-      ? `translateY(${selectedTab.offsetTop + verticalSelectionIndicatorOffset / 2}px)`
-      : `translateX(${selectedTab.offsetLeft}px) `;
-
-    if (orientation === 'horizontal') {
-      styleObj.width = `${selectedTab.offsetWidth}px`;
-    } else {
-      styleObj.height = `${selectedTab.offsetHeight - verticalSelectionIndicatorOffset}px`;
+    if (selectedTab) {
+      let styleObj = {transform: undefined, width: undefined, height: undefined};
+      // In RTL, calculate the transform from the right edge of the tablist so that resizing the window doesn't break the Tabline position due to offsetLeft changes
+      let offset = direction === 'rtl' ? -1 * (selectedTab.offsetParent.offsetWidth - selectedTab.offsetWidth - selectedTab.offsetLeft) : selectedTab.offsetLeft;
+      styleObj.transform = orientation === 'vertical'
+        ? `translateY(${selectedTab.offsetTop + verticalSelectionIndicatorOffset / 2}px)`
+        : `translateX(${offset}px)`;
+
+      if (orientation === 'horizontal') {
+        styleObj.width = `${selectedTab.offsetWidth}px`;
+      } else {
+        styleObj.height = `${selectedTab.offsetHeight - verticalSelectionIndicatorOffset}px`;
+      }
+      setStyle(styleObj);
     }
-    setStyle(styleObj);
-  }, [direction, setStyle, selectedTab, orientation, scale]);
+
+  }, [direction, setStyle, selectedTab, orientation, scale, verticalSelectionIndicatorOffset, selectedKey]);
 
   return <div className={classNames(styles, 'spectrum-Tabs-selectionIndicator')} role="presentation" style={style} />;
 }
+
+function TabsPicker(props) {
+  let {
+    isDisabled,
+    isQuiet,
+    state,
+    'aria-labelledby': ariaLabeledBy,
+    'aria-label': ariaLabel
+  } = props;
+
+  let ref = useRef();
+  let [pickerNode, setPickerNode] = useState(null);
+
+  useEffect(() => {
+    let node = unwrapDOMRef(ref);
+    setPickerNode(node.current);
+  }, [ref]);
+
+  let items = [...state.collection].map((item) => ({
+    rendered: item.rendered,
+    textValue: item.textValue,
+    key: item.key
+  }));
+
+  let pickerProps = {
+    'aria-labelledby': ariaLabeledBy,
+    'aria-label': ariaLabel
+  };
+
+  // TODO: Figure out if tabListProps should go onto the div here, v2 doesn't do it
```

Comment:
I assume stuff like "role: tablist" shouldn't go here so I excluded adding tablistProps from useTabs to the picker

Reply:
cc @majornista